File: java/spring-boot-rabbitmq/spring-boot-rabbitmq-publisher/src/main/java/com/meridal/examples/springbootamqp/repository/RecordingRepository.java
package com.meridal.examples.springbootamqp.repository;

import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Repository;

@Repository
public class RecordingRepository {

    @Value("${messaging.queue}")
    private String queueName;

    private final RabbitTemplate template;

    public RecordingRepository(@Autowired RabbitTemplate template) {
        this.template = template;
    }

    public void publish(String message) {
        this.template.convertAndSend(this.queueName, message);
    }
}

File: kotlin/kotlin-spring-boot-rabbitmq-publisher/src/main/kotlin/com/meridal/examples/kotlinspringbootamqp/repository/RecordingRepository.kt
package com.meridal.examples.kotlinspringbootamqp.repository

import com.meridal.examples.domain.Recording
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Repository

@Repository
class RecordingRepository @Autowired constructor(private val template: RabbitTemplate) {

    @Value("\${messaging.queue}")
    val queueName: String? = null

    fun publish(recording: Recording) {
        template.convertAndSend(queueName!!, recording)
    }
}